src/goto-symex/goto_symex_state.cpp

```diff
@@ -263,24 +263,29 @@ void goto_symex_statet::set_l2_indices(
   ssa_expr.set_level_2(level2.current_count(ssa_expr.get_identifier()));
 }
 
-ssa_exprt
-goto_symex_statet::rename_level0_ssa(ssa_exprt ssa, const namespacet &ns)
+template <goto_symex_statet::levelt level>
+ssa_exprt goto_symex_statet::rename_ssa(ssa_exprt ssa, const namespacet &ns)
 {
-  set_l0_indices(ssa, ns);
-  rename<L0>(ssa.type(), ssa.get_identifier(), ns);
-  ssa.update_type();
-  return ssa;
-}
+  static_assert(
+    level == L0 || level == L1,
+    "rename_ssa can only be used for levels L0 and L1");
+  if(level == L0)
+    set_l0_indices(ssa, ns);
+  else if(level == L1)
+    set_l1_indices(ssa, ns);
+  else
+    UNREACHABLE;
```

Comment:
How about doing a full specialization here? It would do away with the `static_assert` and all the branching (that really isn't one, because the compiler has to remove it anyway). But that's a nit-pick, up to you.

Reply:
This version avoids a bit a code duplication, so I would leave it like that. I think the static assert gives an error message that is clearer than getting a linker error.